src/main/scala/iog/psg/cardano/CardanoApiCodec.scala

```diff
@@ -418,21 +432,25 @@ object CardanoApiCodec {
       }
     }
 
-    def toWallet: Future[CardanoApiResponse[Wallet]]
-    = to[Wallet](Unmarshal(_).to[CardanoApiResponse[Wallet]])
-
-    def toWallets: Future[CardanoApiResponse[Seq[Wallet]]]
-    = to[Seq[Wallet]](Unmarshal(_).to[CardanoApiResponse[Seq[Wallet]]])
-
-
-    def toWalletAddressIds: Future[CardanoApiResponse[Seq[WalletAddressId]]]
-    = to[Seq[WalletAddressId]](Unmarshal(_).to[CardanoApiResponse[Seq[WalletAddressId]]])
-
-    def toFundPaymentsResponse: Future[CardanoApiResponse[FundPaymentsResponse]]
-    = to[FundPaymentsResponse](Unmarshal(_).to[CardanoApiResponse[FundPaymentsResponse]])
+    final def decodeInStream[T](response: HttpResponse, jsonPath: String)(implicit um: Unmarshaller[String, T]): Future[CardanoApiResponse[Seq[T]]] =
+      decodeResponseEntityOrHandleError(response, () =>
+        response.entity.dataBytes
+          .via(JsonReader.select(jsonPath))
+          .mapAsync(4)(bs => unmarshalOrRecoverToUnparseable[T](bs.utf8String))
```

Comment:
put `4` as a val to describe what it is. numberOfParallelFutures

Reply:
`mapAsync(parallelism = 4)` should be ok then